cli/bpmetadata/validate.go

```diff
@@ -27,51 +26,18 @@ func validateMetadata(bpPath, wdPath string) error {
 		bpPath = path.Join(wdPath, bpPath)
 	}
 
-	moduleDirs := []string{bpPath}
-	modulesPath := path.Join(bpPath, modulesPath)
-	_, err := os.Stat(modulesPath)
-	if err == nil {
-		subModuleDirs, err := util.WalkTerraformDirs(modulesPath)
-		if err != nil {
-			Log.Warn("unable to read the submodules i.e. modules/ folder", "err", err)
-		}
-
-		moduleDirs = append(moduleDirs, subModuleDirs...)
+	metadataFiles, err := util.FindFilesWithPattern(bpPath, `.*/metadata(?:.display)?.yaml`)
```

Comment:
Depends on the details, but do you need to escape the `/`?  `\/`

Probably want to anchor the end regardless:

```suggestion
	metadataFiles, err := util.FindFilesWithPattern(bpPath, `.*/metadata(?:.display)?.yaml$`)
```

Reply:
Go doesn't requires you to escape if you provide the regex in back ticks. But anchoring it is valid. I'll do that.